packages/@aws-cdk/aws-ec2/lib/instance-types.ts

```diff
@@ -585,4 +600,22 @@ export class InstanceType {
   public toString(): string {
     return this.instanceTypeIdentifier;
   }
+
+  /**
+   * The instance's CPU architecture
+   */
+  public get architecture(): InstanceArchitecture {
+    // capture the family, generation, and capabilities portion of the instance type
+    const instanceTypePattern = this.instanceTypeIdentifier.match(/^([a-z]+)(\d{1,2})([a-z]*)\..*$/);
```

Comment:
Deal with the possibility of this not matching please. A good rule of thumb when programming with or for other people: "trust nothing" 😄